collections/src/main/scala/scalan/collections/Collections.scala

```diff
@@ -558,6 +552,26 @@ trait CollectionsDsl extends impl.CollectionsAbs with SeqsDsl {
     def flatMap[B: Elem](f: Rep[A] => Coll[B]): Coll[B] = coll.flatMapBy(fun(f))
 
     def :+(x: Rep[A]) = coll.append(x)
+
+    def sort[O: Elem](c: Rep[A] => Rep[O])(implicit o: Ordering[O]) = coll.sortBy(fun(c))
+  }
+
+  implicit class CollectionOfPairExtensions[A, B](coll: Coll[(A, B)])(implicit o: Overloaded1) {
+    implicit lazy val (eItemA, eItemB): (Elem[A], Elem[B]) = coll.selfType1.eItem match {
+      case PairElem(eFst, eSnd) => (eFst, eSnd)
+    }
+    private def pc(coll: Coll[(A, B)]) = PairCollectionAOS(coll)
+    private def self: PairColl[A, B] = PairCollectionAOS(coll)
+
+    def innerMult(other: Coll[(A, B)])(implicit ordK: Ordering[A], nA: Numeric[A], nB: Numeric[B]) = {
+      self.innerJoin[B, B](pc(other), (b1: Rep[(B, B)]) => b1._1 * b1._2)
```

Comment:
Seems I wasn't very accurate in my statement. `innerJoin` is a method of trait `PairColl` and accepts argument of `PairColl` type. And it's better to have `innerMult` accepting and extending `Coll[(A,B)]` rather than `PairColl[A, B]` (thus it won't require explicit conversions while calling it with `Coll[(A,B)]`).
So, here we define "wrappers" for joins in `PairColl[A, B]` as extension of `Coll[(A,B)]`.